lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -681,15 +681,63 @@ class ConvertAtenReductionOp : public OpConversionPattern<AtenOpT> {
                                          keepDims)))
       return failure();
 
+    Value ordValue;
+    if constexpr (std::is_same_v<AtenOpT, AtenLinalgVectorNormOp>) {
+      Type elemType = outputTy.getElementType();
+      if (!elemType.isa<mlir::FloatType>())
+        return rewriter.notifyMatchFailure(
+            op, "Only floating-point datatype legalization supported for "
+                "AtenLinalgVectorNormOp op");
+
+      if (failed(torchScalarToTosaTensor(rewriter, op, op.getOrd(), ordValue,
+                                         elemType, {})))
+        return rewriter.notifyMatchFailure(
+            op, "Currently only scalar constants are supported for "
+                "conversion in TOSA operation");
+
+      // TODO: Add support for ord = {0, +inf, -inf}.
+      auto epsilon = 1e-5;
+      auto ordLiteral = 0.0;
+      if (matchPattern(ordValue, m_TorchConstantFloat(&ordLiteral)) &&
+          fabs(ordLiteral) < epsilon)
+        return rewriter.notifyMatchFailure(op, "unimplemented: L0 norm");
+
+      if (std::isinf(ordLiteral))
+        return rewriter.notifyMatchFailure(op, "unimplemented: ord = +/- inf");
+
+      auto absVal = rewriter.create<tosa::AbsOp>(
+          op.getLoc(),
+          OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+              self.getType()),
+          self);
+      auto powVal = rewriter.create<tosa::PowOp>(
+          op.getLoc(),
+          OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+              self.getType()),
+          absVal, ordValue);
+      self = powVal;
+    }
```

Comment:
Why is this in `ConvertAtenReductionOp` and not in say `ConvertAtenMultipleDimsReductionOp` (which is what we use to instantiate the convertor for this op) or a new one for this op, say `ConvertAtenLinalgVectorNormOp`?

Reply:
Since you are special casing for `AtenLinalgVectorNormOp` anyways here, it seems having a separate convertor for this op would be better. 

@ramiro050 Do you have any recommendations for this?

Reply:
I think we just need to define a `conversionFuncT` for the full norm reduction, and pass that as the template parameter rather than `mlir::tosa::convertReduceSumOp`. It seems to me that that was the original intention with this template pattern.

Reply:
Replace the specialization with `mlir::tosa::convertLinalgVectorNormOp` function

Reply:
@zezhang I think what @ramiro050 was referring to was to define a new reduction function for norm, something like `convertReduceNormOp` that will replace `convertReduceSumOp` used here: https://sourcegraph.com/github.com/llvm/torch-mlir/-/blob/lib/Conversion/TorchToTosa/TorchToTosa.cpp?L5111

Reply:
Yes, that's what I did. `mlir::tosa::convertLinalgVectorNormOp` is the new `conversionFuncT` added in  `TosaLegalizeCommon.cpp` so that the original `ConvertAtenReductionOp` template remain unchanged.

Reply:
Oh yeah. I misread your changes earlier. Thanks for clarifying. That matches my understanding as well.